OpenEdition Journals.js

```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
+			for (let i = 0, n = trans.length; i < n; i++) {
 				item.creators.push(
 					ZU.cleanAuthor(trans[i].textContent, 'translator', true));
 			}
-			//fix all caps for author last names
-			for (var i=0; i<item.creators.length; i++){
-				if (item.creators[i].lastName == item.creators[i].lastName.toUpperCase()){
-					item.creators[i].lastName = ZU.capitalizeTitle(item.creators[i].lastName.toLowerCase(), true)
+			// fix all caps for author last names
+			for (let i = 0; i < item.creators.length; i++) {
+				if (item.creators[i].lastName == item.creators[i].lastName.toUpperCase()) {
+					item.creators[i].lastName = ZU.capitalizeTitle(item.creators[i].lastName.toLowerCase(), true);
 				}
 			}
-			//set abstract and keywords based on preferred locale
-			var locale = doc.cookie.match(/\blanguage=([a-z]{2})/i);
-			//default to french if not set
-			locale = locale ? locale[1].toLowerCase() : 'fr';
+			// set abstract and keywords based on documents language
+			var locale = ZU.xpathText(doc, '//meta[@name="citation_language"]/@content');
+			// default to french if not set
+			locale = locale ? locale.toLowerCase() : 'fr';
 
-			//get abstract  and tags in preferred locale
-			//or the first locale available
+			// get abstract and tags in document language
+			// or the first locale available
 			item.abstractNote = ZU.xpathText(doc,
-				'//meta[@name="description" or @name="DC.description"]\
-						[lang("' + locale + '") or @lang="' + locale + '"][1]\
-						/@content') ||
-				ZU.xpathText(doc,
-					'//meta[@name="description" or @name="DC.description"][1]\
-						/@content');
+				'//meta[@name="description" or @name="DC.description"][lang("' + locale + '") or @lang="' + locale + '"][1]/@content')
+				|| ZU.xpathText(doc,
+					'//meta[@name="description" or @name="DC.description"][1]/@content');
 
 			var tags = ZU.xpathText(doc,
-				'//meta[@name="keywords" or @name="DC.subject"]\
-						[lang("' + locale + '") or @lang="' + locale + '"][1]\
-						/@content') ||
-				ZU.xpathText(doc,
-					'//meta[@name="keywords" or @name="DC.subject"][1]\
-						/@content');
+				'//meta[@name="keywords" or @name="DC.subject"][lang("' + locale + '") or @lang="' + locale + '"][1]/@content')
+				|| ZU.xpathText(doc,
+					'//meta[@name="keywords" or @name="DC.subject"][1]/@content');
 			if (tags) {
 				item.tags = tags.trim().split(/\s*,\s*/);
 			}
 
 			delete item.extra;
 		
-			//The site lists all editor of journals as editor in the header. Turn them into contributors. 
-			//I don't think there is a use case for editors for journal articles
-			if (item.itemType === "journalArticle"){
-				for (i in item.creators){
-					if (item.creators[i].creatorType === "editor"){
+			// The site lists all editor of journals as editor in the header. Turn them into contributors.
+			// I don't think there is a use case for editors for journal articles
+			if (item.itemType === "journalArticle") {
+				for (let i in item.creators) {
```

Comment:
don't use `for ... in` to iterate through arrays. `for ... of` is preferred, explicitly increasing the index as you do elsewhere is also fine.